public/app/core/components/TransformersUI/stretchFrames/stretchFrames.test.ts

```diff
@@ -0,0 +1,151 @@
+import { toDataFrame, ArrayVector, DataFrame, FieldType } from '@grafana/data';
+import { stretchFrames } from './stretchFrames';
+
+describe('Stretch frames transformer', () => {
+  it('should stretch wide to long', () => {
+    const source = [
+      toDataFrame({
+        name: 'wide',
+        refId: 'A',
+        fields: [
+          { name: 'time', type: FieldType.time, values: [10, 9, 8, 7, 6, 5] },
+          { name: 'text', type: FieldType.string, values: ['a', 'z', 'b', 'x', 'c', 'b'] },
+          { name: 'count', type: FieldType.string, values: [1, 2, 3, 4, 5, 6] },
+        ],
+      }),
+    ];
+
+    expect(stretchFrames(source)).toEqual([
+      toEquableDataFrame({
+        name: 'wide0',
+        refId: 'A0',
+        fields: [
+          { name: 'time', type: FieldType.time, values: [10, 9, 8, 7, 6, 5] },
+          { name: 'text', type: FieldType.string, values: ['a', 'z', 'b', 'x', 'c', 'b'] },
```

Comment:
I don't think we want to keep the string field -- old panels only expect numbers (that was all we could make)